lib/trino-record-decoder/src/main/java/io/trino/decoder/protobuf/ProtobufValueProvider.java

```diff
@@ -139,6 +142,15 @@ private static Slice getSlice(Object value, Type type, String columnName)
             return Slices.wrappedBuffer(((ByteString) value).toByteArray());
         }
 
+        if (type.getBaseName().equals(JSON) && value instanceof DynamicMessage) {
```

Comment:
I personally would prefer to IF on the source type ( whether it's https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#value )

but i don't know Kafka connector conventions